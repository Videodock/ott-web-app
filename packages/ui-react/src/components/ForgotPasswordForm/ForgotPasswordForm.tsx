import React from 'react';
import { useTranslation } from 'react-i18next';
import type { ForgotPasswordFormData } from '@jwp/ott-common/types/account';
import type { FormErrors } from '@jwp/ott-common/types/form';
import { testId } from '@jwp/ott-common/src/utils/common';

import Button from '../Button/Button';
import TextField from '../TextField/TextField';
import FormFeedback from '../FormFeedback/FormFeedback';

import styles from './ForgotPasswordForm.module.scss';

type Props = {
  onSubmit: React.FormEventHandler<HTMLFormElement>;
  onChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  onBlur: React.FocusEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  error?: string;
  errors: FormErrors<ForgotPasswordFormData>;
  value: ForgotPasswordFormData;
  submitting: boolean;
};

const ForgotPasswordForm: React.FC<Props> = ({ onSubmit, onChange, value, errors, submitting, onBlur }: Props) => {
  const { t } = useTranslation('account');

  return (
    <form onSubmit={onSubmit} data-testid={testId('forgot-password-form')} noValidate className={styles.forgotPasswordForm}>
      <h2 className={styles.title}>{t('reset.forgot_password')}</h2>
      {errors.form ? <FormFeedback variant="error">{errors.form}</FormFeedback> : null}
      <p className={styles.text}>{t('reset.forgot_text')}</p>
      <TextField
        value={value.email}
        onChange={onChange}
        onBlur={onBlur}
        label={t('reset.email')}
        placeholder={t('reset.email')}
        error={!!errors.email || !!errors.form}
        helperText={errors.email}
        name="email"
        type="email"
        autoComplete="email"
        required
      />
      <Button type="submit" className={styles.button} fullWidth color="primary" disabled={submitting} label={t('reset.email_me')} />
    </form>
  );
};

export default ForgotPasswordForm;
