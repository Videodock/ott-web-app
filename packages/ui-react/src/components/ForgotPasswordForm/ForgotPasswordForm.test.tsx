import React from 'react';
import { axe } from 'vitest-axe';
import { render } from '@testing-library/react';

import ForgotPasswordForm from './ForgotPasswordForm';

describe('<ForgotPasswordForm>', () => {
  test('renders and matches snapshot type forgot', () => {
    const { container } = render(
      <ForgotPasswordForm submitting={false} onSubmit={vi.fn()} onChange={vi.fn()} value={{ email: '' }} errors={{}} onBlur={vi.fn()} />,
    );

    expect(container).toMatchSnapshot();
  });

  test('WCAG 2.2 (AA) compliant', async () => {
    const { container } = render(
      <>
        <h2>Initial state</h2>
        <ForgotPasswordForm submitting={false} onSubmit={vi.fn()} onChange={vi.fn()} value={{ email: '' }} errors={{}} onBlur={vi.fn()} />
        <h2>Error state</h2>
        <ForgotPasswordForm
          submitting={false}
          onSubmit={vi.fn()}
          onChange={vi.fn()}
          value={{ email: '' }}
          errors={{ form: 'Generic error message', email: 'Email error message' }}
          onBlur={vi.fn()}
        />
      </>,
    );

    expect(await axe(container, { runOnly: ['wcag21a', 'wcag21aa', 'wcag22aa'] })).toHaveNoViolations();
  });
});
