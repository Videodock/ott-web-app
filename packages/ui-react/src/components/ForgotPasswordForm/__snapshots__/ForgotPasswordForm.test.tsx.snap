// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ForgotPasswordForm> > renders and matches snapshot type forgot 1`] = `
<div>
  <form
    data-testid="forgot-password-form"
    novalidate=""
  >
    <h2
      class="_title_27331a"
    >
      reset.forgot_password
    </h2>
    <p
      class="_text_27331a"
    >
      reset.forgot_text
    </p>
    <div
      class="_formField_86f2f5 _textField_e16c1b"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_email"
      >
        reset.email
      </label>
      <div
        class="_container_859b20"
      >
        <input
          aria-describedby="helper_text_1235_email"
          aria-invalid="false"
          aria-required="true"
          class="_input_859b20"
          id="text-field_1235_email"
          name="email"
          placeholder="reset.email"
          required=""
          type="email"
          value=""
        />
      </div>
    </div>
    <button
      aria-disabled="false"
      class="_button_f8f296 _button_27331a _primary_f8f296 _outlined_f8f296 _fullWidth_f8f296"
      type="submit"
    >
      <span>
        reset.email_me
      </span>
    </button>
  </form>
</div>
`;
